app/src/main/java/org/schabi/newpipe/RouterActivity.java

```diff
@@ -425,54 +428,79 @@ public void handleChoice(Choice choice) {
             }
         }
 
-        public Consumer<Info> getResultHandler(Choice choice) {
+        public String video_player_key = "video_player_key";
+        public String background_player_key = "background_player_key";
+        public String popup_player_key = "popup_player_key";
+        public boolean isExtVideoEnabled = false;
+        public boolean isExtAudioEnabled = false;
+        public boolean useOldVideoPlayer = false;
+
+
+        public Consumer<Info> getResultHandler(Choice choice, CodeCoverage... codeCoverage) {
+            CodeCoverage cc = codeCoverage != null ? codeCoverage[0] : new CodeCoverage("getResultHandler");
+             String data = "Choice: " + choice;
             return info -> {
-                final String videoPlayerKey = getString(R.string.video_player_key);
-                final String backgroundPlayerKey = getString(R.string.background_player_key);
-                final String popupPlayerKey = getString(R.string.popup_player_key);
+                final String videoPlayerKey = video_player_key;
+                final String backgroundPlayerKey = background_player_key;
+                final String popupPlayerKey = popup_player_key;
 
-                final SharedPreferences preferences = PreferenceManager.getDefaultSharedPreferences(this);
-                boolean isExtVideoEnabled = preferences.getBoolean(getString(R.string.use_external_video_player_key), false);
-                boolean isExtAudioEnabled = preferences.getBoolean(getString(R.string.use_external_audio_player_key), false);
-                boolean useOldVideoPlayer = PlayerHelper.isUsingOldPlayer(this);
+                //final SharedPreferences preferences = PreferenceManager.getDefaultSharedPreferences(this);
+                //boolean isExtVideoEnabled = preferences.getBoolean(getString(R.string.use_external_video_player_key), false);
+                //boolean isExtAudioEnabled = preferences.getBoolean(getString(R.string.use_external_audio_player_key), false);
+                //boolean useOldVideoPlayer = PlayerHelper.isUsingOldPlayer(this);
 
                 PlayQueue playQueue;
                 String playerChoice = choice.playerChoice;
 
                 if (info instanceof StreamInfo) {
+                    cc.visitBranch(0, data);
                     if (playerChoice.equals(backgroundPlayerKey) && isExtAudioEnabled) {
-                        NavigationHelper.playOnExternalAudioPlayer(this, (StreamInfo) info);
+                        cc.visitBranch(1, data);
+                        //NavigationHelper.playOnExternalAudioPlayer(this, (StreamInfo) info);
 
                     } else if (playerChoice.equals(videoPlayerKey) && isExtVideoEnabled) {
```

Comment:
Missed branch as there are two conditions

Reply:
Yeah, but if playerChoice.equals(backgroundPlayerKey), then it will go to branch 6. Do you think I need to add on more branch anyways? Same goes for the others as well.

Reply:
Yeah but they are not in the same control flow, but it doesnt matter really.